src/components/views/admin/SyncIcon.tsx

```diff
@@ -0,0 +1,61 @@
+import { useTranslation } from 'react-i18next';
+
+import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
+import CloudDoneIcon from '@mui/icons-material/CloudDone';
+import CloudOffIcon from '@mui/icons-material/CloudOff';
+import CloudSyncIcon from '@mui/icons-material/CloudSync';
+import { Tooltip } from '@mui/material';
+
+import { TEXT_ANALYSIS } from '@/langs/constants';
+
+type Props = {
+  isSuccess: boolean;
+  isError: boolean;
+  isLoading: boolean;
+  lastSavedMsg?: string;
+};
+
+const ICON_COLOR = '#BBB';
+
+const SyncIcon = ({
+  isSuccess,
+  isError,
+  isLoading,
+  lastSavedMsg,
+}: Props): JSX.Element => {
+  const { t } = useTranslation();
+
+  switch (true) {
```

Comment:
interesting choice to have it inside a switch(true) instead of a series of if statements. Not sure which one is best. lets keep it that way maybe.